File: frontend/apps/test-runner/src/app/api/models/finish-exam-output.ts
/* tslint:disable */
/* eslint-disable */
import { AggregatedGradingOuput } from './aggregated-grading-ouput';
import { QuestionOutput } from './question-output';
import { TimeSpan } from './time-span';
export interface FinishExamOutput {
  examAnswers?: null | {
[key: string]: Array<string>;
};
  finalMark?: number;
  finishedAt?: string;
  grading?: null | Array<AggregatedGradingOuput>;
  questionScores?: null | {
[key: string]: number;
};
  questions?: null | Array<QuestionOutput>;
  totalTime?: TimeSpan;
}

File: backend/VietGeeks.TestPlatform.TestRunner/Contract/ProctorExamActor/FinishExamOutput.cs
namespace VietGeeks.TestPlatform.TestRunner.Contract.ProctorExamActor;

public class FinishExamOutput
{
    public decimal FinalMark { get; set; }
    public List<AggregatedGradingOuput> Grading { get; set; } = default!;
    public DateTime FinishedAt { get; set; }
    public TimeSpan TotalTime { get; set; }
    public IEnumerable<QuestionOutput> Questions { get; set; }
    public IDictionary<string, string[]> ExamAnswers { get; set; }

    public IDictionary<string, int> QuestionScores { get; set; } = default!;
}

public class AggregatedGradingOuput
{
    public int GradingType { get; set; }
    public PassMarkGradeOutput? PassMarkGrade { get; set; }
    public IDictionary<string, string>? Grades { get; set; }
}

public class PassMarkGradeOutput
{
    public bool? IsPass { get; set; }
    public decimal? FinalPoints { get; set; }
    public decimal? TotalPoints { get; set; }
    public decimal? PassValue { get; set; }
    public int Unit { get; set; }
}

public class QuestionOutput
{
    public string Id { get; set; }
    public int QuestionNo { get; set; }
    public string Description { get; set; }
    public string CategoryId { get; set; }
    public int AnswerType { get; set; }
    public AnswerOutput[] QuestionAnswers { get; set; }
    public int TotalPoints { get; set; }
}

public class AnswerOutput
{
    public string Id { get; set; } = default!;
    public string AnswerDescription { get; set; } = default!;
    public bool IsCorrect { get; set; }
}